component/src/main/java/org/wso2/extension/siddhi/io/report/util/QueryModeDataProvider.java

```diff
@@ -0,0 +1,147 @@
+ */
+
+package org.wso2.extension.siddhi.io.report.util;
+
+import ar.com.fdvs.dj.domain.builders.ColumnBuilder;
+import ar.com.fdvs.dj.domain.builders.DynamicReportBuilder;
+import ar.com.fdvs.dj.domain.entities.columns.AbstractColumn;
+import com.zaxxer.hikari.HikariDataSource;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.wso2.siddhi.core.exception.SiddhiAppRuntimeException;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * This class is the implementation of the query mode data provider.
+ */
+public class QueryModeDataProvider implements DataProvider {
+    private static Logger logger = Logger.getLogger(QueryModeDataProvider.class);
+    private HikariDataSource dataSource;
+    private String dataSourceName;
+    private List<AbstractColumn> abstractColumns;
+    private Map<String, AbstractColumn> abstractColumnMap;
+    private DynamicReportBuilder reportBuilder;
+
+    public QueryModeDataProvider(String dataSourceName) {
+        this.dataSourceName = dataSourceName;
+        this.dataSource = RDBMSUtil.getDataSourceService(this.dataSourceName);
+    }
+
+    public void setReportBuilder(DynamicReportBuilder reportBuilder) {
+        this.reportBuilder = reportBuilder;
+    }
+
+    private Connection getConnection() {
+        Connection conn;
+        try {
+            conn = this.dataSource.getConnection();
+        } catch (SQLException e) {
+            throw new SiddhiAppRuntimeException("Cannot create connection from datasource '" + this.dataSourceName +
+                    "'.", e);
+        }
+        return conn;
+    }
+
+    public List<Map<String, Object>> processData(String query) {
+        Connection conn = this.getConnection();
+        PreparedStatement stmt = null;
+        ResultSet resultSet = null;
+        ResultSetMetaData metaData = null;
+        List<Map<String, Object>> data = new ArrayList<>();
+        this.abstractColumns = new ArrayList<>();
+        this.abstractColumnMap = new HashMap<>();
+
+        try {
+            stmt = conn.prepareStatement(query);
+            resultSet = stmt.executeQuery();
+            metaData = resultSet.getMetaData();
+            while (resultSet.next()) {
+                Map<String, Object> dataMap = new HashMap<>();
+                for (int i = 0; i < metaData.getColumnCount(); i++) {
+                    dataMap.put(metaData.getColumnName(i + 1), resultSet.getObject(i + 1));
+                }
+                data.add(dataMap);
+            }
+            addAbstractColumns(metaData);
+            return data;
+        } catch (SQLException e) {
+            throw new SiddhiAppRuntimeException("Cannot retrieve records from  datasource '" + this.dataSourceName + "'.", e);
```

Comment:
The line length exceeds the limit of 120

Reply:
Fixed with f4ca5e3e8c58212e300a47df07b127fd5f818fd7